tests/test_integration.py

```diff
@@ -1,39 +1,51 @@
 # Copyright 2016 Canonical Limited.  All rights reserved.
 
-from fixtures import TestWithFixtures
-
 from twisted.internet import reactor
-from twisted.trial.unittest import TestCase
 from twisted.internet.defer import inlineCallbacks
 
-from txjuju.api import Endpoint, Juju1APIClient
+from txjuju.api import Endpoint, JujuAPIClient
 from txjuju.errors import APIAuthError, APIRequestError
-from txjuju.testing.fakejuju import FakeJujuFixture
 
+from testresources import ResourcedTestCase
+
+from testtools import TestCase
+from testtools.twistedsupport import AsynchronousDeferredRunTest
+
+from fakejuju.fixture import MODEL_UUID
+
+from tests.resources import fakejuju
 
-class Juju1APIClientIntegrationTest(TestCase, TestWithFixtures):
+
+class JujuAPIClientIntegrationTest(TestCase, ResourcedTestCase):
+
+    resources = [("fakejuju", fakejuju)]
+
+    run_tests_with = AsynchronousDeferredRunTest.make_factory(timeout=60)
 
     @inlineCallbacks
     def setUp(self):
-        super(Juju1APIClientIntegrationTest, self).setUp()
-        self.juju = self.useFixture(FakeJujuFixture())
-        self.endpoint = Endpoint(
-            reactor, str(self.juju.address), Juju1APIClient)
+        super(JujuAPIClientIntegrationTest, self).setUp()
+        cli = self.fakejuju.cli()
+        cli.execute("bootstrap", "foo", "bar")
+        self.addCleanup(cli.execute, "destroy-controller", "-y", "bar")
+
+        addr = "localhost:%d" % (self.fakejuju.port - 1)
```

Comment:
I've changed to use a convenience API provided by fakejuju itself: FakeJuju.address.